Mateusz santander

MateuszHaberla opened the pull request:
Userdetails działa. Znaczy tak mi się wydaje. Błędy nie lecą :)
Zmieniłęm w UserdetailsServiceImpl. Zamiast Userrepo używamy tam teraz Userservice.

Commit:
test commiit

Commit:
Merge branch 'master' of https://github.com/WojciechKlups/w-sumie-drogoV2FINAL into Mateusz-Santander

Commit:
Application properties w pracy mi nie działa. Dodałem konfgurację wysyłania maili jako klase konfiguracyjną.

Małe poprawy i refactoring

Commit:
Application properties w pracy mi nie działa. Dodałem konfgurację wysyłania maili jako klase konfiguracyjną.

Małe poprawy i refactoring

Zamiana userrepo na userservice w userdetails

Commit:
trying to handle userdto userdetailserviceimpl

Commit:
changed on userservice instead of userrepo in userdetailserviceimpl

wersun commented on src/main/java/pl/sda/wsumiedrogo/controllers/AccountsController.java:
```diff
@@ -15,20 +16,23 @@
     @GetMapping("/account")
     public String getUserByEmail(HttpServletResponse response, @RequestParam String email, Model model, @ModelAttribute User user) {
-        return accountService.getAccount(userService, model, email, cookieService, user, response);
+        return accountService.getAccount(userService, model, email, cookieService, user, response,userMapper);
```
Raczej serwisow nie przekazuj w argumentach metody tylko sobie je wstrzyknij w serwisie w ktorym tych zaleznosci potrzebujesz.

MateuszHaberla replied:
Poprawione :)

wersun commented on src/main/java/pl/sda/wsumiedrogo/controllers/AuthenticationController.java:
```diff
@@ -9,6 +9,7 @@
 @Controller
 public class AuthenticationController {
 
+
```
Takiego czegos nie commitujemy, tylko zaciemnia Pull Requesta .

wersun commented on src/main/java/pl/sda/wsumiedrogo/controllers/RegisterController.java:
```diff
@@ -32,7 +32,7 @@ public RegisterController(WebSecurityConfig webSecurityConfig, UserService userS
     @GetMapping("/register")
     @ResponseStatus(HttpStatus.CREATED)
```
Dlaczego tutaj zwracamy kod 201 jak w metodzie zwracamy widok ?

wersun commented on src/main/java/pl/sda/wsumiedrogo/service/AccountService.java:
```diff
@@ -11,8 +12,11 @@
     public String getAccount(UserService userService, Model model,
-                             String email, CookieService cookieService, User user, HttpServletResponse response) {
-        UserDto userDto = userService.getUserByEmail(email);
+                             String email, CookieService cookieService, User user,
```
Tak jak pisalem wyzej zamiast przekazywac potrzebne zaleznosci w argumentach lepiej je wstrzyknac tutaj w klasie.

wersun commented on src/main/java/pl/sda/wsumiedrogo/service/UserService.java:
```diff
@@ -28,9 +28,8 @@ public UserDto getUserById(Long id) {
     }
 
-    public UserDto getUserByEmail(String email) {
+    public User getUserByEmail(String email) {
```
w serwisie lepiej juz zwracac obiekt DTO, jest jakis powod dla tego ?

MateuszHaberla replied:
Jutro zrobię refactor na Dto. :)

wersun commented on src/main/java/pl/sda/wsumiedrogo/service/LoginService.java:
```diff
@@ -10,15 +12,16 @@
 
 
-    public String isLoggedIn(HttpServletRequest request,String username,
+    public String isLoggedIn(HttpServletRequest request, String username,
```
Tutaj tak jak Ci mowilem, nie powinienes zwracac nazwy widoku w serwisie, pomijajac to taka metoda nie bedzie potrzebna jak bedzie poprawnie zrobione security.

MateuszHaberla replied:
Nie wiem jak to ugryźć. Principal jest nullem. :)

wersun commented on src/main/java/pl/sda/wsumiedrogo/service/AccountService.java:
```diff
@@ -11,8 +12,11 @@
+        User userByEmail = userService.getUserByEmail(email);
+
+        UserDto userDto = userMapper.userToUserDto(userByEmail);
 //        UserDetails userDetails = userDetailsService.loadUserByUsername(email);
```
Nie commitujemy komentarzy !!!

Commit:
Refactoring
Trying to fix all mistakes. Still some problems with UserService method which shpould take as a parameter UserDto.
Principal is returning null after login. Have no idea why and how to fix it.
